src/server/pfs/server/source.go

```diff
@@ -27,11 +27,22 @@ type source struct {
 }
 
 // NewSource creates a Source which emits FileInfos with the information from commit, and the entries return from fileSet.
-func NewSource(commitInfo *pfs.CommitInfo, fileSet fileset.FileSet, full bool) Source {
+func NewSource(storage *fileset.Storage, commitInfo *pfs.CommitInfo, fs fileset.FileSet, opts ...SourceOption) Source {
```

Comment:
This information could easily be communicated through a wrapped fileset.FileSet passed in the constructor.  None of the new state is used by any of sources methods.

I understand the goal of this is to consolidate the index resolver and dir inserter.  That can be done with a helper function that takes these additional arguments, without adding complexity to Source.

Reply:
Discussed offline. We are sticking with the code as is to guarantee that the dir inserter is setup and to capture the full configuration in one place.